samples/DevTestLabs/Modules/Library/Az.DevTestLabs2.psm1

```diff
@@ -0,0 +1,2636 @@
+          $p | Add-Member -Name 'disallowPublicIpAddress' -Value $True -MemberType NoteProperty
+        }
+
+        if($CustomImage) {
+          if($CustomImage -is [string]) {
+            $SubscriptionID = (Get-AzureRmContext).Subscription.Id
+            $imageId = "/subscriptions/$SubscriptionID/ResourceGroups/$ResourceGroupName/providers/Microsoft.DevTestLab/labs/$Name/customImages/$CustomImage"
+            Write-Verbose "Using custom image (string) $CustomImage with resource id $imageId"
+          } elseif($CustomImage.ResourceId) {
+            $imageId = $CustomImage.ResourceId
+            Write-Verbose "Using custom image (object) $CustomImage with resource id $imageId"
+          } else {
+            throw "CustomImage $CustomImage is not a string and not an object with a ResourceId property."
+          }
+          $p | Add-Member -Name 'customImageId' -Value $imageId -MemberType NoteProperty
+        }
+
+        if($Sku -or $Publisher -or $Offer) {
+          $g = New-Object Object
+          $g | Add-member -Name 'Sku' -Value $Sku -MemberType NoteProperty
+          $g | Add-member -Name 'OsType' -Value $OsType -MemberType NoteProperty
+          $g | Add-member -Name 'Publisher' -Value $Publisher -MemberType NoteProperty
+          $g | Add-member -Name 'Version' -Value $Version -MemberType NoteProperty
+          $g | Add-member -Name 'Offer' -Value $Offer -MemberType NoteProperty
+
+          $p | Add-member -Name 'galleryImageReference' -Value $g -MemberType NoteProperty
+        }
+
+        $jsonToDeploy = ConvertTo-Json $t -Depth 6 # Depth defaults to 2, so it wouldn't print out the GalleryImageReference Property. Why oh why ...
+        $deploymentName = "$Name-$VmName"
+        Write-debug "JSON ABOUT TO BE DEPLOYED AS $deploymentName `n$jsonToDeploy"
+
+        $jsonPath = StringToFile($jsonToDeploy)
+
+        Write-Verbose "Starting deployment $deploymentName of $VmName in $Name"
+
+        $sb = {
+          param($deploymentName, $Name, $ResourceGroupName, $VmName, $jsonPath, $justAz)
+
+          if($justAz) {
+            Enable-AzureRmAlias -Scope Local -Verbose:$false
+          }
+          $deployment = New-AzureRmResourceGroupDeployment -Name $deploymentName -ResourceGroupName $ResourceGroupName -TemplateFile $jsonPath -existingLabName $Name -newVmName $VmName
+          Write-debug "Deployment succeded with deployment of `n$deployment"
+
+          Get-AzureRmResource -Name "$Name/$VmName" -ResourceGroupName $ResourceGroupName -ExpandProperties
+        }
+
+        if($AsJob.IsPresent) {
+          Start-Job      -ScriptBlock $sb -ArgumentList $deploymentName, $Name, $ResourceGroupName, $vmName, $jsonPath, $justAz
+        } else {
+          Invoke-Command -ScriptBlock $sb -ArgumentList $deploymentName, $Name, $ResourceGroupName, $vmName, $jsonPath, $justAz
+        }
+      }
+    } catch {
+      Write-Error -ErrorRecord $_ -EA $callerEA
+    }
+  }
+  end {}
+}
+
+## Experimental ...
```

Comment:
e81cd33 changed to comment to explain what it does, instead of 'experimental'. It is not strictly a DTL thing, but useful often in a DTL context. I am not sure about leaving it in or not. Happy to remove it if you feel better to do so.

## Not strictly DTL related, but often used in a DTL context if unused RGs are left around by broken DTL remove operations.

Reply:
As long as it is clear, we can leave it. No harm.